### PR TITLE
Reuse pre-commit checks in CI

### PR DESCRIPTION
**Pull request by sfdye**
Continuation of #1231 

- [x] Merge black/isort/flake8/mypy into one testenv `lint`
- [x] Re-use pre-commit checks in tox (if any fails locally it should as well fail in CI) (ideas from [here](https://github.com/sloria/environs/blob/master/tox.ini) and [here](https://github.com/asottile/pyupgrade/blob/master/tox.ini))

Tested locally
```
$ tox
  lint: commands succeeded
  py35: commands succeeded
  py36: commands succeeded
  py37: commands succeeded
  py38: commands succeeded
  docs: commands succeeded
  congratulations :)
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,8 +16,3 @@ repos:
     rev: 3.7.9
     hooks:
       - id: flake8
-  - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
-    hooks:
-      - id: end-of-file-fixer
-      - id: trailing-whitespace
```
Comment (sfdye) [inline .pre-commit-config.yaml]: These are no pretty noisy and not really used, thus removed

File: tox.ini [hunk 2]
```diff
@@ -1,41 +1,24 @@
 [tox]
 envlist =
+    lint,
     py{35,36,37,38},
-    flake8,
-    isort,
-    black
+    docs
```
Comment (sfdye) [inline tox.ini]: I think we should test docs build in CI

**Comment by sfdye** [comment]
image

Travis failing because this PR removes several testenv, the build should be green after merging to master.

File: .isort.cfg [hunk 3]
```diff
@@ -4,5 +4,5 @@ include_trailing_comma=True
 force_grid_wrap=0
 use_parentheses=True
 line_length=88
-known_third_party=deprecated,httpretty,jwt,requests,setuptools,urllib3
+known_third_party=deprecated,httpretty,jwt,pytest,requests,setuptools,urllib3
```
Comment (sfdye) [inline .isort.cfg]: Auto-fixed by `seed-isort-config`

**Comment by s-t-e-v-e-n-k** [comment]
Er, and indeed, why aren't you updating .travis.yml in this PR?

**Comment by sfdye** [comment]
@s-t-e-v-e-n-k Good catch

**Comment by sfdye** [comment]
All green now 🤦‍♂

**Comment by s-t-e-v-e-n-k** [comment]
@sfdye I shouldn't tease you on a Monday morning before coffee? :laughing:

**Comment by sfdye** [comment]
> Further, github/Consts.py github/Deployment.py and github/TimelineEventSource.py are missing type hints, did you want to cover that in this PR or do it in a later PR?

Maybe next PR, at least mypy checks can pass now

**Comment by sfdye** [comment]
> @sfdye I shouldn't tease you on a Monday morning before coffee? 😆

This is what happens when you commit code on a Sunday night at 11pm. 

(Making my first coffee of the week after I post this comment)

**Comment by s-t-e-v-e-n-k** [comment]
Finish off your coffee, update one line in CONTRIBUTING.md, and we can merge this. :-)

**Comment by sfdye** [comment]
@s-t-e-v-e-n-k Done

**Comment by s-t-e-v-e-n-k** [comment]
:ship:

**Pull request merged**